### PR TITLE
feat(reindex): Extend instance-records reindex endpoint with passing index settings

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Extend instance-records reindex endpoint with passing index settings

### Approach
- Extend full/upload reindex endpoints with index settings
- Recreate indices on publishing merge ranges instead of doing this on upload start in order to use index settings
- Add instance reindexing to tenant service
- Remove instance record type from old reindex endpoint

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [x] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MSEARCH-853](https://folio-org.atlassian.net/browse/MSEARCH-853)

### Learning and Resources (if applicable)
During full reindex, index recreation is not longer part of upload stage. Indices get recreated after publishing records ranges in order to pass index settings.
During upload reindex, index recreation logic stays intact

### CODE REVIEW

**Comment by viacheslavkol** [comment]
Verified on rancher environment. Full/upload endpoints accept and act on passed index settings

File: README.md [hunk 1]
```diff
@@ -323,12 +323,13 @@ x-okapi-token: [JWT_TOKEN]
 
 {
   "recreateIndex": true,
-  "resourceName": "instance"
+  "resourceName": "authority"
 }
 ```
 
-* `resourceName` parameter is optional and equal to `instance` by default. Possible values: `instance`, `authority`, `location`,
```
Comment (psmagin) [inline README.md:330]: Does it make sense to make the parameter required without default? 
For me it's strange to have some default for it, what do you think?

File: src/main/java/org/folio/search/service/reindex/ReindexService.java [hunk 2]
```diff
@@ -94,20 +96,34 @@ public CompletableFuture<Void> submitFullReindex(String tenantId) {
...
+    log.info("submitUploadReindex:: for [tenantId: {}, entities: {}]", tenantId, entityTypes);
 
-    for (var reindexEntityType : reindexEntityTypes) {
+    validateUploadReindex(tenantId, entityTypes);
+
+    for (var reindexEntityType : entityTypes) {
       statusService.recreateUploadStatusRecord(reindexEntityType);
-      reindexCommonService.recreateIndex(reindexEntityType, tenantId);
+      if (recreateIndex) {
+        reindexCommonService.recreateIndex(reindexEntityType, tenantId, indexSettings);
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/reindex/ReindexService.java:121]: What is the case for not recreating indexes?

File: README.md (hunk 1 above)
Comment (viacheslavkol) [reply README.md:330]: Sound good to make it required

File: src/main/java/org/folio/search/service/reindex/ReindexService.java (hunk 2 above)
Comment (viacheslavkol) [reply src/main/java/org/folio/search/service/reindex/ReindexService.java:121]: When it's full reindex. In such case - index settings are passed on merge phase so now indexes are recreated on merge ranges published in order to use passed settings. Upload phase in full flow is triggered on kafka events consuming and so index is not recreated there now

**Review by psmagin — APPROVED** [review]

**Review by SvitlanaKovalova1 — APPROVED** [review]

**Pull request merged**